anthos-multi-cloud/customize-logs-fluentbit/kubernetes/fluentbit-configmap.yaml

```diff
@@ -0,0 +1,171 @@
+        storage.backlog.mem_limit  50M
+    @INCLUDE input-containers.conf
+    @INCLUDE filter-kubernetes.conf
+    @INCLUDE output-stackdriver.conf
+  input-containers.conf: |
+    [INPUT]
+        # https://docs.fluentbit.io/manual/input/tail
+        # https://docs.fluentbit.io/manual/pipeline/filters/kubernetes#workflow-of-tail-kubernetes-filter
+        Name               tail
+        Tag                k8s_application.<namespace_name>.<pod_name>.<container_name>
+        Tag_Regex          var.log.containers.(?<pod_name>[a-z0-9](?:[-a-z0-9]*[a-z0-9])?(?:\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace_name>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\.log$
+        Path               /var/log/containers/*.log
+        DB                 /var/log/fluent-bit-user-k8s-container-application.db
+        Buffer_Chunk_Size  512KB
+        Buffer_Max_Size    2MB
+        Rotate_Wait        30
+        # Limit of memory that Tail plugin can use when appending data to the engine.
+        Mem_Buf_Limit      50MB
+        Skip_Long_Lines    On
+        Refresh_Interval   5
+        storage.type       filesystem
+        Read_from_Head     True
+    [FILTER]
+        # https://docs.fluentbit.io/manual/pipeline/filters/modify
+        Name          modify
+        Match         k8s_application*
+        Hard_rename   log message
+    # Parses the workload log lines and tries parsers one after another.
+    # cri,glog and json are applied in order until one succeeds.
+    [FILTER]
+        Name         parser
+        Match        k8s_application*
+        Key_Name     message
+        Reserve_Data True
+        Parser       cri
+        Parser       appglog
+        Parser       json
+    [FILTER]
+        # https://docs.fluentbit.io/manual/pipeline/filters/modify
+        Name         modify
+        Match        k8s_application*
+        Copy         level severity
+    [FILTER]
+        # https://docs.fluentbit.io/manual/pipeline/filters/modify
+        Name         modify
+        Match        k8s_application*
+        Condition    Key_value_equals stream stdout
+        Add          severity I
+    [FILTER]
+        # https://docs.fluentbit.io/manual/pipeline/filters/modify
+        Name         modify
+        Match        k8s_application*
+        Condition    Key_value_equals stream stderr
+        Add          severity E
+  replace_info.lua: |
+        function replace_sensitive_info(tag, timestamp, record)
+          -- mask social security number
+          record["message"] = string.gsub(record["message"], "%d%d%d%-+%d%d%-+%d%d%d%d", "xxx-xx-xxxx")
+          -- mask credit card number
+          record["message"] = string.gsub(record["message"], "%d%d%d%d *%d%d%d%d *%d%d%d%d *%d%d%d%d", "xxxx xxxx xxxx xxxx")
+          -- mask email address
+          record["message"] = string.gsub(record["message"], "[%w+%.%-_]+@[%w+%.%-_]+%.%a%a+", "user@email.tld")
+          return 1, timestamp, record
+        end
```

Comment:
nit: maybe this should be remarked as well, given that L121 comments script execution?

Reply:
I'd prefer to leave this section uncommented so the user only has to uncomment one section, since there are already several steps involved with turning on log customization